consensus/types/src/chain_spec.rs

```diff
@@ -865,6 +865,28 @@ impl ChainSpec {
         )
     }
 
+    /// Returns the slot at which the proposer shuffling was decided.
+    ///
+    /// The block root at this slot can be used to key the proposer shuffling for the given epoch.
+    pub fn proposer_shuffling_decision_slot<E: EthSpec>(&self, epoch: Epoch) -> Slot {
```

Comment:
At `epoch == fulu_fork_epoch`, I think this function returns incorrect decision slot, as the epochs proposer shuffling decision slot should be the slot at the end of epoch N - 1 right?

Reply:
Addressed in #8101